packages/babel-types/src/validators/isReferenced.ts

```diff
@@ -44,31 +44,34 @@ export default function isReferenced(
     case "ClassMethod":
     case "ClassPrivateMethod":
     case "ObjectMethod":
-      // @ts-expect-error todo(flow->ts) params have more specific type comparing to node
-      if (parent.params.includes(node)) {
-        return false;
```

Comment:
Before this PR we checked `key` and `params` and return true if `node` is neither in `params` nor in `key`. But the `return true` branch is unreachable, unless the input `node` does not have parental relationships with `parent`.